halo/evmstaking2/keeper/keeper_internal_test.go

```diff
@@ -95,13 +153,57 @@ func setupKeeper(t *testing.T, submissionDelay int64) (*Keeper, sdk.Context) {
 
 	k, err := NewKeeper(
 		storeSvc,
-		nil,
-		nil,
-		nil,
-		nil,
+		ethCl,
+		aKeeper,
+		bKeeper,
+		sKeeper,
+		msgServer,
 		submissionDelay,
 	)
 	require.NoError(t, err, "new keeper")
 
 	return k, ctx
 }
+
+type stakingKeeperMock struct{}
```

Comment:
tiny nit:
- I like to differentiate between mocks and stubs.
- Mock are complex things that assert specific behaviour and mimic logic.
- Stubs are minimal-just-dont-panic things that dont assert and most just returns happy path results.

These would mostly be stubs in that case, not mocks.